app/assets/javascripts/angular/controller/home-controller.js.erb

```diff
@@ -1,17 +1,50 @@
-app.controller('HomeCtrl', ['$scope', '$http', function($scope, $http){
+app.controller('HomeCtrl', ['$scope', '$http', '$routeParams', '$location', function($scope, $http, $routeParams, $location){
 
-  $http.get('pins.json').success(function(data){
-    console.log('sucess on getting pins from HomeCtrl');
-    // console.log(data);
-    $scope.pins = data.pins;
-    var numberOfPins = data.pins.length;
-    // console.log('from controller I see', angular.element('.box').length,'boxes');
-  });
+  $scope.searchTerms = '';
+  $scope.category = '';
+  var categoryId;
+  var url = '';
 
+  $scope.search = function() {
+    if ($scope.category) { 
+      categoryId = $scope.category.id;
+      url = 'pins.json?keyword=' + $scope.searchTerms +'&category_id=' + categoryId;
```

Comment:
* Why do you need ```categoryId```, would it be easier to directly do ```$scope.category.id``` in your url?
* Don't you need ```/``` before ```pins.json```?